android.py

```diff
@@ -100,44 +76,51 @@ def setup_android_target_template(env: dict, opts: AndroidOpts, target: str):
             'android-x86_64_CXXFLAGS': ['-DL_cuserid=9']
         }
     }
-
-    if target in extra_target_envs:
-        env.update(extra_target_envs[target])
-
-    arch = AndroidTargetTable.archs[target]
-    abi_name = AndroidTargetTable.abi_names[target]
-    host_triple = AndroidTargetTable.host_triples[target]
-    api = env['ANDROID_API_VERSION']
-
-    toolchain_path = path_join(opts.android_toolchains_prefix, opts.toolchain_name_fmt % (target, api))
-
-    tools_path = path_join(toolchain_path, 'bin')
-    name_fmt = abi_name + '-%s'
-
-    sdk_cmake_dir = path_join(opts.android_sdk_root, 'cmake', get_android_cmake_version(opts))
-    if not os.path.isdir(sdk_cmake_dir):
-        print('Android CMake directory \'%s\' not found' % sdk_cmake_dir)
-
-    AR = path_join(tools_path, name_fmt % 'ar')
-    AS = path_join(tools_path, name_fmt % 'as')
-    CC = path_join(tools_path, name_fmt % 'clang')
-    CXX = path_join(tools_path, name_fmt % 'clang++')
+    env.update(extra_target_envs[target])
+
+    if target == "armv7":
+        target_triple = "armv7a-linux-androideabi"
+        bin_utils = "arm-linux-androideabi"
+    elif target == "arm64v8":
+        target_triple = "aarch64-linux-android"
+        bin_utils = target_triple
+    elif target == "x86":
+        target_triple = "i686-linux-android"
+        bin_utils = target_triple
+    elif target == "x86_64":
+        target_triple = "x86_64-linux-android"
+        bin_utils = target_triple
+
+    if sys.platform.startswith("linux"):
+        host_subpath = "linux-x86_64"
+    elif sys.platform.startswith("darwin"):
+        host_subpath = "darwin-x86_64"
+    elif sys.platform.startswith("win"):
+        if platform.machine().endswith("64"):
+            host_subpath = "windows-x86_64"
+        else:
+            host_subpath = "windows"
+
+    cmake_path = os.path.join(opts.android_sdk_root, 'cmake', opts.android_cmake_version, 'bin')
+    ndk_path = os.path.join(opts.android_sdk_root, 'ndk', opts.android_ndk_version)
+    toolchain_path = os.path.join(ndk_path, 'toolchains/llvm/prebuilt', host_subpath)
+    compiler_path = os.path.join(toolchain_path, 'bin')
+    compiler_wrapper = target_triple + env['ANDROID_API_VERSION'] + '-'
+    bin_utils_path = os.path.join(toolchain_path, bin_utils, 'bin')
+    android_api = env['ANDROID_API_VERSION']
+
+    AR = os.path.join(compiler_path, 'llvm-ar')
+    AS = os.path.join(bin_utils_path, 'as')
```

Comment:
As seen in https://github.com/godotengine/godot/pull/62813, using `clang` directly seems more reliable than taking `as` from `bin_utils_path`, so you should be able to replicate the same change here and drop `bin_utils` and `bin_utils_path`.

Reply:
Using `clang` for compiling assembly code allows for the pre-processing required for the new `zstd` `huf_decompress_amd64.S` file. However, it breaks the `libvpx` build (removed in 4.0), because it can't compile the `libvpx` assembly code. So godotengine/godot#62813 can't currently be backported to 3.x or here.

Reply:
Alright, that probably indicates a bug in our libvpx build setup (which is a huge mess), but that should be handled later on, for now it's better to stay compatible with `3.x`.

Reply:
```
/usr/local/lib/android/sdk/ndk/23.2.8568313/toolchains/llvm/prebuilt/linux-x86_64/bin/clang -mfpu=neon -o thirdparty/libvpx/vpx_dsp/arm/gas/intrapred_neon_asm.android.opt.armv7.neon.o thirdparty/libvpx/vpx_dsp/arm/gas/intrapred_neon_asm.s
```

Or it's just overriding `ASFLAGS` somewhere in the `libvpx` config, since `--target` and `-c` are clearly missing from the command.